### ISSUE TITLE
[bug] use `enigo` in tauri cause app crashed

### ISSUE THREAD

**Issue by jackeydou** [tauri-apps/tauri#6421]
### Describe the bug

I try to use `enigo` to do some keyboard event mock in my tauri app, and it will crash when the macOS api `TISCopyCurrentKeyboardInputSource` or `TISGetInputSourceProperty` execute in `enigo`.

I start an [issue](https://github.com/enigo-rs/enigo/pull/161) in enigo before, but we found the crash will only appear in my Tauri app.

### Reproduction

1. register a shortcut
```rust
import { isRegistered, register, unregister } from '@tauri-apps/api/globalShortcut';
export async function registerShortcut(shortcut: string, callback: (shortcut?: string) => void) {
  await register(shortcut, (shortcut) => {
    callback(shortcut);
  });
}
```
and there is a log in shell: 
```shell
register Accelerator { id: Some(AcceleratorId(29420)), mods: ALT, key: KeyS }
```

2. write a tauri command, and invoke in javascript
```rust
#[tauri::command]
async fn paste_content(window: tauri::Window) {
    let mut enigo = Enigo::new();
    let trusted = accessibility::query_accessibility_permissions();
    if trusted {
        enigo.key_down(Key::Meta);
        enigo.key_click(Key::Layout('v'));
        enigo.key_up(Key::Meta);

    }
}
```
```javascript
invoke("paste_content").then(async _ => {
        
 })
```
I use `accessibility` to help me request the macOS accessibility permission, and it returns `true`.
3. the application crashed

### Expected behavior

The app does not crash.

### Platform and versions

```text
Environment
  › OS: Mac OS 13.0.0 X64
  › Node.js: 16.14.0
  › npm: 8.3.1
  › pnpm: 7.9.5
  › yarn: 3.1.1
  › rustup: 1.25.2
  › rustc: 1.68.0
  › cargo: 1.68.0
  › Rust toolchain: stable-aarch64-apple-darwin 

Packages
  › @tauri-apps/cli [NPM]: 1.2.3
  › @tauri-apps/api [NPM]: 1.2.0
  › tauri [RUST]: 1.2.4,
  › tauri-build [RUST]: 1.2.1,
  › tao [RUST]: 0.15.8,
  › wry [RUST]: 0.23.4,

App
  › build-type: bundle
  › CSP: unset
  › distDir: ../dist
  › devPath: http://localhost:1420/
  › framework: React
  › bundler: Vite

App directory structure
  ├─ dist
  ├─ node_modules
  ├─ public
  ├─ .github
  ├─ src-tauri
  ├─ api
  ├─ .git
  ├─ .vscode
  ├─ assets
  └─ src
```

### Stack trace

```text
There is no output in my shell when it crashed, I try to debug it and set some breakpoints, I put the screenshots [here](https://github.com/enigo-rs/enigo/pull/161#issuecomment-1460236620)
```

**Comment by yetone** [comment]
I have encountered the same problem and it is very troubling.

**Comment by jackeydou** [comment]
> I have encountered the same problem and it is very troubling.

Did you find the cause of the crash and how did you fix it?

**Comment by garrrikkotua** [comment]
Have the same problem, though the app doesn't crash in my case

**Comment by garrrikkotua** [comment]
@zeenix @amrbashir could you please take a look? 🙏

**Comment by zeenix** [comment]
@garrrikkotua why me? I don't even know what enigo is.

**Comment by garrrikkotua** [comment]
> @garrrikkotua why me? I don't even know what enigo is.

Apologies, it was a suggestion from GitHub 😅

**Comment by amrbashir** [comment]
I don't have access to a mac unfortunately so I can't triage this issue.

**Comment by parker-codes** [comment]
This is going to sound a bit strange, but try disabling spellcheck in the HTML.

```html
<html spellcheck="false">
    <!-- ... -->
</html>
```

About a year ago this was causing WebKit to crash in our MacOS variant.

**Comment by pentamassiv** [comment]
A user tried it out and the problem persisted

**Comment by uwejan** [comment]
@pentamassiv Is there an update on this.
I notice app exists without any error code.
Disabeling spellchecker does not help.

**Comment by pentamassiv** [comment]
No, sorry I have not looked into it. I am not familiar at all with tauri

**Comment by inomata137** [comment]
I'm using tauri@2 and enigo@0.3.0 .
The problem didn't happen when I called `enigo.key_down(...)` from the main thread.
However, the app crashed when I called from a newly spawned thread, even though `accessibility::query_accessibility_permissions()` returned `true` in the spawned thread.

**Comment by inomata137** [comment]
For me, this helped:
```rust
// this works, but blocks the main thread while sleeping
#[tauri::command]
fn this_works() {
    use enigo::Keyboard;

    let mut enigo = enigo::Enigo::new(&enigo::Settings::default()).unwrap();
    enigo.key(enigo::Key::Unicode('a'), enigo::Direction::Click).unwrap();
    std::thread::sleep(std::time::Duration::from_secs(5));
    enigo.key(enigo::Key::Unicode('b'), enigo::Direction::Click).unwrap();
}

#[tauri::command]
fn this_crashes() {
    std::thread::spawn(|| {
        use enigo::Keyboard;

        let mut enigo = enigo::Enigo::new(&enigo::Settings::default()).unwrap();
        enigo.key(enigo::Key::Unicode('a'), enigo::Direction::Click).unwrap();
        std::thread::sleep(std::time::Duration::from_secs(5));
        enigo.key(enigo::Key::Unicode('b'), enigo::Direction::Click).unwrap();
    });
}

// this works and doesn't block the main thread while sleeping
#[tauri::command]
fn this_also_works(app: AppHandle) {
    std::thread::spawn(move || {
        use enigo::Keyboard;

        app.run_on_main_thread(|| {
            let mut enigo = enigo::Enigo::new(&enigo::Settings::default()).unwrap();
            enigo.key(enigo::Key::Unicode('a'), enigo::Direction::Click).unwrap();
        }).unwrap();
        std::thread::sleep(std::time::Duration::from_secs(5));
        app.run_on_main_thread(|| {
            let mut enigo = enigo::Enigo::new(&enigo::Settings::default()).unwrap();
            enigo.key(enigo::Key::Unicode('b'), enigo::Direction::Click).unwrap();
        }).unwrap();
    });
}

```

**Comment by 0xDAEF0F** [comment]
> For me, this helped:
> 
> // this works, but blocks the main thread while sleeping
> #[tauri::command]
> fn this_works() {
>     use enigo::Keyboard;
> 
>     let mut enigo = enigo::Enigo::new(&enigo::Settings::default()).unwrap();
>     enigo.key(enigo::Key::Unicode('a'), enigo::Direction::Click).unwrap();
>     std::thread::sleep(std::time::Duration::from_secs(5));
>     enigo.key(enigo::Key::Unicode('b'), enigo::Direction::Click).unwrap();
> }
> 
> #[tauri::command]
> fn this_crashes() {
>     std::thread::spawn(|| {
>         use enigo::Keyboard;
> 
>         let mut enigo = enigo::Enigo::new(&enigo::Settings::default()).unwrap();
>         enigo.key(enigo::Key::Unicode('a'), enigo::Direction::Click).unwrap();
>         std::thread::sleep(std::time::Duration::from_secs(5));
>         enigo.key(enigo::Key::Unicode('b'), enigo::Direction::Click).unwrap();
>     });
> }
> 
> // this works and doesn't block the main thread while sleeping
> #[tauri::command]
> fn this_also_works(app: AppHandle) {
>     std::thread::spawn(move || {
>         use enigo::Keyboard;
> 
>         app.run_on_main_thread(|| {
>             let mut enigo = enigo::Enigo::new(&enigo::Settings::default()).unwrap();
>             enigo.key(enigo::Key::Unicode('a'), enigo::Direction::Click).unwrap();
>         }).unwrap();
>         std::thread::sleep(std::time::Duration::from_secs(5));
>         app.run_on_main_thread(|| {
>             let mut enigo = enigo::Enigo::new(&enigo::Settings::default()).unwrap();
>             enigo.key(enigo::Key::Unicode('b'), enigo::Direction::Click).unwrap();
>         }).unwrap();
>     });
> }

 I have a question. Why is there a need to call sleep between one keypress and the other in Enigo?

**Comment by pentamassiv** [comment]
There is no need to call sleep between key presses

**Comment by depressi0n** [comment]
With tauri@2 and enigo@0.5.0, replace enigo::Key::Unicode(<char>) with enigo::Key::Other(<key_code>), there should be a mismatched mapping from specified value to underlying key code, and you are expected to find the interested key code in core-graphics-<some_version>/src/event.rs.
Hope above information is helpful.

**Comment by pentamassiv** [comment]
This should be fixed by https://github.com/enigo-rs/enigo/pull/510
